torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -34,6 +34,7 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
     return {};
 
   auto new_grad = std::move(grads[0]);
+  std::lock_guard<std::mutex> lock(grad_mutex);
```

Comment:
Above this line there is a comment `// XXX: this method is not thread-safe!` that I'm guessing needs updating